### PR TITLE
[ui] Bringing all dependancies inline with each other

### LINKED ISSUE #1591: Various dependancies is `ui` are backlogged or not maintained
**Describe the bug**

Various dependencies is `ui` are backlogged or not maintained. Update them to a new version or a maintained version.

### PR DESCRIPTION
**Pull request by vishakhanihore**
fixes #1591

### CODE REVIEW

**Comment by vishakhanihore** [comment]
@djmitche @helfi92 I made sure to align each `peer-dependancy` as much as possible . Also there were some useless packages which I have removed from `yarn.lock` as they are no longer required by the updated package. Please have a look

**Comment by vishakhanihore** [comment]
Dont review it now, There are some syntax changes after dependency updates. I will push them by tonight and check the status. Thanks

**Comment by vishakhanihore** [comment]
@djmitche @helfi92 Ready for your review 🎉

File: ui/src/components/HookForm/index.jsx [hunk 1]
```diff
@@ -750,9 +750,7 @@ export default class HookForm extends Component {
                             window.env.TASKCLUSTER_ROOT_URL,
                             'reference/core/hooks/firing-hooks'
                           )}>
-                          {'"'}
-                          firing hooks
-                          {'"'}
+                          &apos;&quot;&apos; firing hooks &apos;&quot;&apos;
```
Comment (djmitche) [inline ui/src/components/HookForm/index.jsx]: I think you just want `&quot;` here (with no space)

File: ui/src/components/Homepage/index.jsx [hunk 2]
```diff
@@ -65,9 +65,8 @@ export default class HomePage extends Component {
           </Typography>
         </Hidden>
         <Typography variant="h5">
-          {
-            "A task execution framework that supports Mozilla's continuous integration and release processes."
-          }
+          &quot; A task execution framework that supports Mozilla&apos;s
```
Comment (djmitche) [inline ui/src/components/Homepage/index.jsx]:
```suggestion
          &quot;A task execution framework that supports Mozilla&apos;s
```
and is the `&apos;` really necessary??

File: ui/src/components/Homepage/index.jsx [hunk 3]
```diff
@@ -65,9 +65,8 @@ export default class HomePage extends Component {
...
+          continuous integration and release processes. &quot;
```
Comment (djmitche) [inline ui/src/components/Homepage/index.jsx]:
```suggestion
          continuous integration and release processes.&quot;
```

File: ui/src/components/HookForm/index.jsx (hunk 1 above)
Comment (kushthedude) [reply ui/src/components/HookForm/index.jsx]: Yes

File: ui/src/components/Homepage/index.jsx (hunk 3 above)
Comment (kushthedude) [reply ui/src/components/Homepage/index.jsx]: @vishakhanihore yes but why do you need to change symbols into such string?

File: ui/src/components/Homepage/index.jsx (hunk 2 above)
Comment (vishakhanihore) [reply ui/src/components/Homepage/index.jsx]: Its to differentiate between `Mozillas` and `Mozilla's`

File: ui/src/components/Homepage/index.jsx (hunk 3 above)
Comment (vishakhanihore) [reply ui/src/components/Homepage/index.jsx]: > @vishakhanihore yes but why do you need to change symbols into such string?

Due to the syntax and rule updates.Thanks

File: ui/src/components/Homepage/index.jsx (hunk 2 above)
Comment (djmitche) [reply ui/src/components/Homepage/index.jsx]: I mean instead of just using the `'` character.

File: ui/src/components/HookForm/index.jsx (hunk 1 above)
Comment (vishakhanihore) [reply ui/src/components/HookForm/index.jsx]: Yes you are correct, my bad. Thanks

**Comment by vishakhanihore** [comment]
@djmitche @helfi92 Updated the PR. Have a look.

File: ui/src/components/Homepage/index.jsx (hunk 2 above)
Comment (vishakhanihore) [reply ui/src/components/Homepage/index.jsx]: It threw a error on console .

**Comment by vishakhanihore** [comment]
@djmitche @helfi92 Done the rebase, Please review before more merge conflicts occurs. 😅

File: ui/src/components/Homepage/index.jsx (hunk 3 above)
Comment (djmitche) [reply ui/src/components/Homepage/index.jsx]: It looks like the error was
```
WARNING in ./src/components/Homepage/index.jsx
Module Warning (from ./node_modules/eslint-loader/index.js):
error: Curly braces are unnecessary here (react/jsx-curly-brace-presence) at src/components/Homepage/index.jsx:68:11:
  66 |         </Hidden>
  67 |         <Typography variant="h5">
> 68 |           {
     |           ^
  69 |             "A task execution framework that supports Mozilla's continuous integration and release processes."
  70 |           }
  71 |         </Typography>
```

The original form was providing a JS string, and using `{ .. }` so that we could write JS.  I think the reason for this was to allow use of the `'` character without escaping it.  But a change to eslint in this PR is causing that to no longer be accepted.

The `".."` quotes never appeared onscreen - they were just JS syntax.  So I think the right config here is
```diff
-         &quot;A task execution framework that supports Mozilla&apos;s
-         continuous integration and release processes. &quot;
+         A task execution framework that supports Mozilla&apos;s
+         continuous integration and release processes.
```

File: ui/src/components/HookForm/index.jsx [hunk 4]
```diff
@@ -749,9 +749,7 @@ export default class HookForm extends Component {
                             window.env.TASKCLUSTER_ROOT_URL,
                             'reference/core/hooks/firing-hooks'
                           )}>
-                          {'"'}
-                          firing hooks
-                          {'"'}
+                          &quot;firing hooks&quot;
```
Comment (djmitche) [inline ui/src/components/HookForm/index.jsx]: This one looks OK :)

**Comment by vishakhanihore** [comment]
@djmitche @helfi92 I will commit the sentence change in a succeding PR, Since it has quite a lot of dependancies involved, Every merge results in quite a amount of conflicts and it needs a rebase again. 😅

**Comment by djmitche** [comment]
That is a disadvantage of this kind of project!

**Comment by djmitche** [comment]
I think that setting just disallows merging a PR unless it's up-to-date -- it doesn't automatically merge or rebase.  Am I mistaken?

**Comment by djmitche** [comment]
Cool!  I don't see it in the settings, though -- can you point me to it, or point me to documentation about the future?

**Comment by vishakhanihore** [comment]
You need to go to 'settings->branch', since we have 'master' as the default
branch , implement a protection rule for it, in this you can check to
implement update branch, number of approving reviews before merging, also
an option to discard previous approval if new changes have been made by the
author.

**Comment by djmitche** [comment]
I enabled the feature briefly.  I didn't see an "update branch" button, though.  I do see "Resolve conflicts", but that occurs without the feature enabled, too.  I'm missing something :)

File: ui/package.json [hunk 5]
```diff
@@ -126,5 +129,10 @@
   "optionalDependencies": {
     "bufferutil": "^4.0.0",
     "utf-8-validate": "^5.0.0"
+  },
+  "peerDependencies": {
+    "@material-ui/core": "^1.0.0",
+    "mocha": "^5.0.0",
+    "karma": "^3.0.0"
```
Comment (helfi92) [inline ui/package.json]: Why are we adding these peer dependencies?

File: ui/package.json [hunk 6]
```diff
@@ -109,6 +111,7 @@
     "cross-env": "^6.0.3",
     "dotenv-cli": "^2.0.1",
     "eslint": "^5.6.1",
+    "eslint-plugin-babel": "^5.3.0",
```
Comment (helfi92) [inline ui/package.json]: Why are we adding this dependency?

File: ui/package.json [hunk 7]
```diff
@@ -56,11 +58,11 @@
     "localforage": "^1.7.3",
     "lodash.clonedeep": "^4.5.0",
     "lodash.debounce": "^4.0.8",
-    "markdown-it": "^9.0.0",
+    "markdown-it": "^10.0.0",
     "markdown-it-highlightjs": "^3.0.0",
     "markdown-it-link-attributes": "^2.1.0",
     "material-ui-pickers": "2.2.4",
-    "material-ui-treeview": "^3.2.0",
+    "material-ui-treeview": "^3.4.0",
```
Comment (helfi92) [inline ui/package.json]: 👍

File: ui/package.json [hunk 8]
```diff
@@ -56,11 +58,11 @@
     "localforage": "^1.7.3",
     "lodash.clonedeep": "^4.5.0",
     "lodash.debounce": "^4.0.8",
-    "markdown-it": "^9.0.0",
+    "markdown-it": "^10.0.0",
```
Comment (helfi92) [inline ui/package.json]: 👍

File: ui/package.json [hunk 9]
```diff
@@ -37,6 +38,7 @@
     "change-case": "^3.1.0",
     "classnames": "^2.2.6",
     "codemirror": "^5.39.0",
+    "core-js": "3",
```
Comment (helfi92) [inline ui/package.json]: Why are we adding this dependency?

Comment (kushthedude) [reply ui/package.json]: Reason for core js is during yarn start, it gives you are using corejs 2.6 please upgrade to atleast 3 and build stalls for 3-4 sec at the same warning.

Comment (vishakhanihore) [reply ui/package.json]: Without its presence the build gets stall on the netify as well as on terminal . Its due to the updated `React-Scripts`

File: ui/package.json (hunk 6 above)
Comment (vishakhanihore) [reply ui/package.json]: It might have been added by my local dev environment

File: ui/package.json (hunk 5 above)
Comment (vishakhanihore) [reply ui/package.json]: `mocha` and `karma` are a peer dependency for most of the `dependencies` we are using, This reduces the warnings on console by a significant amount.

**Comment by vishakhanihore** [comment]
@helfi92 Requesting review before arising of more conflicts 😅

File: ui/package.json (hunk 5 above)
Comment (helfi92) [reply ui/package.json]: I'm not sure I understand. They are already specified in the list of dependencies so why list them in peerDependencies as well?

File: ui/package.json (hunk 9 above)
Comment (helfi92) [reply ui/package.json]: Which `React-Scripts`? Do you have a link I can look at or something. Just trying to understand :)

**Comment by vishakhanihore** [comment]
@helfi92 , I will update you by tomorrow and also file PR for the new issue
assigned to me. Busy with University exams for today and tomorrow😅.

**Comment by vishakhanihore** [comment]
@helfi92  regarding the core-js 
```sh
WARNING: We noticed you're using the `useBuiltIns` option without declaring a core-js version. Currently, we assume version 2.x when no version is passed. Since this default version will likely change in future versions of Babel, we recommend explicitly setting the core-js version you are using via the `corejs` option.

You should also be sure that the version you pass to the `corejs` option matches the version specified in your `package.json`'s `dependencies` section. If it doesn't, you need to run one of the following commands:

  npm install --save core-js@2    npm install --save core-js@3
  yarn add core-js@2              yarn add core-js@3
```

File: ui/package.json (hunk 5 above)
Comment (vishakhanihore) [reply ui/package.json]: Fixed it, Actually some of the depdendancies have been missing from `yarn.lock` peers.

**Comment by vishakhanihore** [comment]
@helfi92 Requesting review before arising of more conflicts 😅

**Comment by vishakhanihore** [comment]
@helfi92 @djmitche Requesting your reviews before arising of more conflicts 😅

File: ui/package.json (hunk 6 above)
Comment (helfi92) [reply ui/package.json]: I'm not sure we need this dependency.

File: ui/package.json [hunk 10]
```diff
@@ -24,6 +24,7 @@
     "@mdx-js/runtime": "^1.0.20",
     "@mozilla-frontend-infra/components": "^2.4.0",
     "@sentry/browser": "^5.0.0",
+    "@types/react": "16.8.0",
```
Comment (helfi92) [inline ui/package.json]: Why is this needed?

Comment (vishakhanihore) [reply ui/package.json]: It was auto-generated, After I upgraded `Yarn`. 
Also after using this `yarn.lock` and `package.json `, Number of warnings have been reduced to **1**

File: ui/package.json (hunk 6 above)
Comment (vishakhanihore) [reply ui/package.json]: I will remove it once the other dependency queries are resolved 😄 , Its just a dependency for `local development environment`.
Also I think should we include a section for `preferable development environment` for **TaskCluster**, Like using the react debugger and setting up `React-EsLint`.

File: ui/package.json (hunk 10 above)
Comment (helfi92) [reply ui/package.json]: `yarn.lock` is autogenerated but I don't think `ui/package.json` is. You can try doing `yarn remove @types/react`.

File: ui/package.json (hunk 6 above)
Comment (vishakhanihore) [reply ui/package.json]: @helfi92 Should I remove this too ? As it will be used to solve local lint errors ?

**Comment by vishakhanihore** [comment]
@helfi92 Please Review

File: ui/src/components/HookForm/index.jsx [hunk 11]
```diff
@@ -748,9 +748,7 @@ export default class HookForm extends Component {
...
```
Comment (helfi92) [inline ui/src/components/HookForm/index.jsx]: This was changed recently in https://github.com/taskcluster/taskcluster/commit/3da927a303f5e7062780f71bea6648f74626ccb9. The quotations were removed so this won't be needed anymore.

File: ui/src/components/Homepage/index.jsx [hunk 12]
```diff
@@ -65,9 +65,8 @@ export default class HomePage extends Component {
           </Typography>
         </Hidden>
         <Typography variant="h5">
-          {
-            "A task execution framework that supports Mozilla's continuous integration and release processes."
-          }
+          &quot;A task execution framework that supports Mozilla&apos;s
+          continuous integration and release processes. &quot;
```
Comment (helfi92) [inline ui/src/components/Homepage/index.jsx]: This was fixed in https://github.com/taskcluster/taskcluster/commit/3da927a303f5e7062780f71bea6648f74626ccb9. The latest master branch should be good.

File: ui/src/components/Dashboard/DocsSidebarList.jsx [hunk 13]
```diff
@@ -263,7 +263,9 @@ export default class DocsSidebarList extends Component {
               component={Link}
               to={removeReadmeFromPath(item.path)}
               classes={{ container: classes.listItem }}>
-              <ListItemIcon>{<item.icon />}</ListItemIcon>
+              <ListItemIcon>
+                <item.icon />
+              </ListItemIcon>
```
Comment (helfi92) [inline ui/src/components/Dashboard/DocsSidebarList.jsx]: Looks like this was fixed in https://github.com/taskcluster/taskcluster/commit/3da927a303f5e7062780f71bea6648f74626ccb9.

File: ui/package.json (hunk 6 above)
Comment (helfi92) [reply ui/package.json]: OK, I think we can remove it now.

**Comment by vishakhanihore** [comment]
@helfi92 Made your requested changes.

**Comment by vishakhanihore** [comment]
Please Review @djmitche @helfi92

File: ui/yarn.lock [hunk 14]
```diff
@@ -17,6 +17,8 @@
   dependencies:
     "@apollo/react-common" "^3.1.3"
```
Comment (helfi92) [inline ui/yarn.lock]: The tests are failing. I wonder if it's because there's something wrong with the `yarn.lock` file. You can try running `yarn` inside the `ui` directory to see if there are any changes.

Comment (helfi92) [reply ui/yarn.lock]: When removing packages, `yarn.lock` usually updates when the package is removed via `yarn remove <package-name>`.

**Comment by vishakhanihore** [comment]
Will solve it right away.

**Comment by vishakhanihore** [comment]
Closed in favour of #1718
